pkg/controller/storagecluster/reconcile.go

```diff
@@ -262,19 +254,28 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		r.ensureCephConfig,
 		r.ensureCephCluster,
 		r.ensureNoobaaSystem,
-	} {
+	}
+	// for external cluster, we have a different set of ensure functions
+	if instance.Spec.ExternalStorage.Enable {
+		ensureFs = []ensureFunc{
+			r.ensureCephConfig,
+			r.ensureCephCluster,
+		}
+	}
+	for _, f := range ensureFs {
 		err = f(instance, reqLogger)
-		if r.phase == statusutil.PhaseClusterExpanding {
+		if r.phase == statusutil.PhaseClusterExpanding && !instance.Spec.ExternalStorage.Enable {
```

Comment:
```suggestion
		if r.phase == statusutil.PhaseClusterExpanding {
```

If it is PhaseClusterExpanding, it is  definitely not an External StorageCluster.
Instead of doing this, we should ensure we don't set the wrong phase to the wrong StorageCluster and infer the rest.

Reply:
Please take a look at the above change in the code, we tried to handle wrong phase issue.